authorization/http.go

```diff
@@ -35,6 +35,15 @@ func WithAuthorizers(authorizers map[string]rbac.Authorizer, permission rbac.Per
 				return
 			}
 
+			token, ok := authentication.GetAccessToken(r.Context())
+			if !ok {
+				http.Error(w, "error finding access token", http.StatusUnauthorized)
+
+				return
+			}
+
+			a.SetAccessToken(token)
```

Comment:
This is _very_ dangerous. The authorizer is used highly concurrently as the API handles many requests in parallel. This line, coupled with the fact that there is no synchronization in setting the token and accessing the authorizer, means that we have no way of knowing who's token is being forwarded for a given request.

Imagine request A comes into the API. This causes the code above to set A's token on the authorizer. 1 millisecond later request B arrives and B's token is set on the authorizer. Next, the authorization middleware makes post requests to the OPA authorizer and sends B's token along for both A and B. This means that A will be authorized according to B's credentials.